feat(swarm): make stream uprade errors more ergonomic (#3882)
paritytech/rust-libp2p: swarm/CHANGELOG.md

  This variant was never constructed and thus dead code.
  See [PR 3605].

> - Deprecate `ConnectionHandlerUpgrErr` in favor of `StreamUpgradeError`.
>   This new type features a flat interface which should be easier to use.
>   See [PR 3882].

Reviewer: Changelog reads like the `ConnectionHandlerUpgrErr` doesn't change. Though given that it is a type alias, it is flattened just like `StreamUpgradeError`.

Would something along the lines below help?
```suggestion
- Flatten `ConnectionHandlerUpgrErr` and rename to `StreamUpgradeError`.
  See [PR 3882].
```

Author: Sounds good to me!

>   See [PR XXXX].
> 
> [PR 3605]: https://github.com/libp2p/rust-libp2p/pull/3605
> [PR 3746]: https://github.com/libp2p/rust-libp2p/pull/3746
> [PR 3715]: https://github.com/libp2p/rust-libp2p/pull/3715
> [PR 3746]: https://github.com/libp2p/rust-libp2p/pull/3746
> [PR 3865]: https://github.com/libp2p/rust-libp2p/pull/3865
> [PR XXXX]: https://github.com/libp2p/rust-libp2p/pull/XXXX

Author: ```suggestion
  See [PR 3882].

[PR 3605]: https://github.com/libp2p/rust-libp2p/pull/3605
[PR 3746]: https://github.com/libp2p/rust-libp2p/pull/3746
[PR 3715]: https://github.com/libp2p/rust-libp2p/pull/3715
[PR 3746]: https://github.com/libp2p/rust-libp2p/pull/3746
[PR 3865]: https://github.com/libp2p/rust-libp2p/pull/3865
[PR 3882]: https://github.com/libp2p/rust-libp2p/pull/3882
```
